deploy/provision/mesh-egress-ab.sh

```
@@ -0,0 +1,181 @@
+
+# --- auto A/B across approved exit nodes (needs tailscale CLI) ------------------
+do_ab(){
+  need curl; need awk; need tailscale
+  local orig nodes
+  orig=$(tailscale debug prefs 2>/dev/null | awk -F'"' '/"ExitNodeID"/{print $4}')
+  restore(){ [ -n "$orig" ] && tailscale set --exit-node="$orig" --exit-node-allow-lan-access >/dev/null 2>&1 || tailscale set --exit-node= >/dev/null 2>&1 || true; }
+  trap restore EXIT
+  nodes=$(tailscale exit-node list 2>/dev/null | awk 'NR>1 && $2 ~ /\./ {print $2}')
+  [ "$JSON" = 1 ] && echo "["
+  LABEL="direct (no exit node)"; tailscale set --exit-node= >/dev/null 2>&1 || true; sleep 3
+  [ "$JSON" = 1 ] && do_measure | sed 's/$/,/' || { echo "=== DIRECT (no exit node) ==="; do_measure; }
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)